flox-bash/lib/utils.sh

```diff
@@ -1438,10 +1438,12 @@ function identifyParentShell() {
 	# Only attempt a guess if we know our parent PID.
 	if [ -n "${FLOX_PARENT_PID:-}" ]; then
 		# First attempt to identify details of parent shell process.
-		if [ -L "/proc/$FLOX_PARENT_PID/exe" -a \
-			 -r "/proc/$FLOX_PARENT_PID/exe" ]; then
-			# Linux - use information from /proc.
-			parentShell="$($_readlink "/proc/$FLOX_PARENT_PID/exe")"
+		if [ -r "/proc/$FLOX_PARENT_PID/cmdline" ]; then
+			# Linux - use information from /proc. bash can be invoked as sh, so we use
+			# argv[0] rather than /proc/$FLOX_PARENT_PID/exe
+			# -d '' splits on null bytes
+			mapfile -d '' cmdline < <(cat "/proc/$FLOX_PARENT_PID/cmdline")
+			parentShell="${cmdline[0]}"
```

Comment:
(are we using `cat` or `$_cat`?)

`/proc/$FLOX_PARENT_PID/cmdline` contains the entire command line string separated by NUL. Is `mapfile -d ''` doing a similar thing as this?

Leave it to you to accept this suggestion if its equivalent.


```suggestion
		if [ -r "/proc/$FLOX_PARENT_PID/cmdline" ]; then
			# Linux - use information from /proc. bash can be invoked as sh, so we use
			# argv[0] rather than /proc/$FLOX_PARENT_PID/exe
	        # cmdline is NUL separated, so we pass it through `strings` to extract the 
	        # first component 
			parentShell="$($_cat "/proc/$FLOX_PARENT_PID/cmdline" | $_strings -n 1)"
```

Reply:
Yep good catch on `cat`

mapfile splits on the separator specified by `-d` and puts it in an array. And then I'm taking the 0th element. Doesn't strings output all the strings in a file? So I don't understand how your suggestion gets just the 0th element?